### DISCUSSION TITLE
Placing scenery in viewport does not rotate to match preview

### DISCUSSION THREAD

**Discussion by spacek531** [OpenRCT2/OpenRCT2#21762] [Feature requests]
When using extra viewport that is rotated differently than the main view, the scenery is rotated to match the scenery selection window based on the main viewport. It would be useful if scenery rotated to match the scenery selection window in whatever window the mouse is currently over.

![image](https://github.com/OpenRCT2/OpenRCT2/assets/3454156/d17dfbba-0bbf-45f6-ba92-4c08832d1cf6)

The scenery is rotated differently than the preview, when using extra viewport.

**Comment by ZehMatt** [comment]
The scenery rotation is its own thing so its tied to that and not the viewport rotation, I'm guessing the extra viewport uses a different rotation than your primary viewport? I'm not sure how we could properly implement that, we would probably have to strip the scenery rotation entirely and have the user do it with viewport rotation instead, which doesn't seem that practical to me.

**Reply by spacek531** [reply]
I have not looked at the scenery placement code at all, but it seems logical to me that when placing the scenery, the game looks at the main viewport's rotation and adds that to the scenery selection window's rotation. If that is the case, then it's a matter of creating a new property that sets the viewport rotation that the user is mousing over, and scenery placement uses that.

**Reply by ZehMatt** [reply]
I don't think that is an intuitive thing, the viewport is just that, an extra viewport with its own rotation, some people might not want this so this would require an extra option as well.

**Reply by spacek531** [reply]
When I am using extra viewport to build, I have to constantly remember that the rotation in the scenery selection window is wrong. In this example, I am using the extra viewport to build a sphynx in a place I cannot see. The normal scan pattern is:
1. identify the need for a scenery piece in a specific spot by looking at the spot (in the extra viewport)
2. find the scenery in the scenery selection window and rotate it to the desired rotation (based on the information from step 1 - from the extra viewport rotation)
3. try to place the scenery in the specific spot with the desired rotation. The scenery is rotated incorrectly.
![image](https://github.com/OpenRCT2/OpenRCT2/assets/3454156/a21eb1dd-9f29-4601-823b-791a335656cc)

Using extra viewport to look at two sides the same object and trying to build is disorienting because I have to rotate the object constantly. On large objects you place very few of, the issue is manageable, but for small scenery you place a lot of, the rotation issue is annoying and slows building.

**Reply by ZehMatt** [reply]
What is your proposal? There are a couple of challenges to add the extra viewport rotation to the object rotation without having a ton of special code in a lot of places.

**Reply by spacek531** [reply]
When placing objects (and object ghosts), the objects should match the rotation selected in the scenery window. As far as recall, the only types of objects that can be placed with rotation are small scenery, large scenery, banners, and ride track.

Regarding option to disable/enable, I don't think it needs to be a game setting. It can be added to the viewport itself, if necessary:

![image](https://github.com/OpenRCT2/OpenRCT2/assets/3454156/785e0f05-9bfe-41bf-9c1f-09c7cb24eef0)
